gapir/client/device_connection.go

```diff
@@ -384,72 +332,16 @@ func (s *session) newADB(ctx context.Context, d adb.Device, abi *device.ABI, lau
 			return true, nil
 		})
 	if err != nil {
-		return log.Errf(ctx, err, "Checking socket: %v", socketPath)
+		return nil, log.Errf(ctx, err, "Checking socket: %v", socketPath)
 	}
 	log.I(ctx, "Gapir socket: '%v' is opened now", socketPath)
 
 	if err := d.Forward(ctx, localPort, adb.NamedFileSystemSocket(socketPath)); err != nil {
-		return log.Err(ctx, err, "Forwarding port")
-	}
-	s.onClose(func() { d.RemoveForward(ctx, localPort) })
-
-	log.I(ctx, "Waiting for connection to GAPIR...")
-	s.conn, err = newConnection(fmt.Sprintf("localhost:%d", localPort), s.auth, connectTimeout)
-	if err != nil {
-		return log.Err(ctx, err, "Timeout waiting for connection")
+		return nil, log.Err(ctx, err, "Forwarding port")
 	}
-	s.onClose(func() {
-		if s.conn != nil {
-			s.conn.Close()
-		}
-	})
-	log.I(ctx, "Heartbeat connection setup done")
-	return nil
-}
 
-func (s *session) connect(ctx context.Context) (gapir.Connection, error) {
-	<-s.inited
-	return newConnection(fmt.Sprintf("localhost:%d", s.port), s.auth, connectTimeout)
-}
-
-func (s *session) onClose(f func()) {
-	s.closeCBs = append(s.closeCBs, f)
-}
-
-func (s *session) close(ctx context.Context) {
-	if s.conn != nil {
-		s.conn.Shutdown(ctx)
-	}
-	for _, f := range s.closeCBs {
-		f()
-	}
-	s.closeCBs = nil
-}
-
-func (s *session) ping(ctx context.Context) (time.Duration, error) {
-	if s.conn == nil {
-		return time.Duration(0), log.Errf(ctx, nil, "cannot ping without gapir connection")
-	}
-	start := time.Now()
-	err := s.conn.Ping(ctx)
-	if err != nil {
-		return 0, err
-	}
-	return time.Since(start), nil
-}
-
-func (s *session) heartbeat(ctx context.Context, pingInterval time.Duration) {
-	for {
-		select {
-		case <-task.ShouldStop(ctx):
-			return
-		case <-time.After(pingInterval):
-			_, err := s.ping(ctx)
-			if err != nil {
-				log.E(ctx, "Error sending keep-alive ping. Error: %v", err)
-				s.close(ctx)
-				return
-			}
-		}
-	}
+	return &deviceConnectionInfo{port: port, authToken: "", cleanupFunc: func() {
+		cleanup.Invoke(ctx)
+		d.RemoveForward(ctx, localPort)
```

Comment:
Please declare cleanupFunc outside of the struct field initializer, as you did earlier in your refactoring:
```
cleanupFunc := func() { ... }
...
return &deviceConnectionInfo{..., cleanupFunc: cleanupFunc}
```

Reply:
Also, is there a good reason why we don't use authToken for ADB connections?

Reply:
I don't see a reason for the first comment.

Reply:
For the second question, I do not know. I have not touched the internals of the implementations. I cannot give any information but it was never set to any value.

Reply:
OK for authToken, let's keep it to `""`.

For cleanupFunc, I would prefer to have it defined outside because:
- this is how it is done in the other similar `newXXX` functions
- having the lambda defined directly here reduces code readability, the `return` line becomes quite complex

Reply:
We should be generating an auth-token for Android as well. I am not sure why we are not, but we absolutely should be.
```
authToken := auth.GenToken()
``` 
in newADB should do it.